File: src/dataAccessServer/Repositories/Logic/WorkoutDAO.cs
using Repositories.Interfaces;
using Shared.DTOs;

namespace Repositories.Logic;

public class WorkoutDAO : IWorkoutDAO
{

    public Task<Workout> GetWorkout(int id)
    {
        return Task.FromResult(new Workout
        {
            Id = 1
        });
    }

    public Task<IEnumerable<Workout>> GetWorkouts()
    {
        throw new NotImplementedException();
    }

    public Task<FollowWorkoutDTO> AssignWorkout(FollowWorkoutDTO dto)
    {
        // Write impl

        return Task.FromResult(dto);
    }

    public Task<Workout> EditWorkout(Workout workout)
    {
        // Write impl

        return Task.FromResult(workout);
    }

    public Task<bool> DeleteWorkout(int id)
    {
        // Write impl

        return Task.FromResult(true);
    }
}

File: src/logicServer/src/main/java/via/sep3/logicserver/repositories/interfaces/WorkoutDAO.java
package via.sep3.logicserver.repositories.interfaces;

import via.sep3.logicserver.shared.FollowWorkoutDTO;
import via.sep3.logicserver.shared.WorkoutModel;
import via.sep3.logicserver.shared.WorkoutDTO;

import java.util.List;

public interface WorkoutDAO {

    WorkoutModel getWorkout(int id) throws Exception;

    List<WorkoutModel> GetWorkouts() throws Exception;

    FollowWorkoutDTO assignWorkout(FollowWorkoutDTO dto) throws Exception;

    void editWorkout(WorkoutModel workout) throws Exception;

    void deleteWorkout(int id) throws Exception;

    WorkoutDTO createWorkout(WorkoutDTO workoutDTO) throws Exception;
}
